inst/templates/cran/msrv.R

```diff
@@ -0,0 +1,116 @@
+# read the DESCRIPTION file
+desc <- read.dcf("DESCRIPTION")
+
+if (!"SystemRequirements" %in% colnames(desc)) {
+  fmt <- c(
+    "`SystemRequirements` not found in `DESCRIPTION`.",
+    "Please specify `SystemRequirements: Cargo (Rust's package manager), rustc`"
+  )
+  stop(paste(fmt, collapse = "\n"))
+}
+
+# extract system requirements
+sysreqs <- desc[, "SystemRequirements"]
+
+# check that cargo and rustc is found
+if (!grepl("cargo", sysreqs, ignore.case = TRUE)) {
+  stop("You must specify `Cargo (Rust's package manager)` in your `SystemRequirements`")
+}
+
+if (!grepl("rustc", sysreqs, ignore.case = TRUE)) {
+  stop("You must specify `Cargo (Rust's package manager), rustc` in your `SystemRequirements`")
+}
+
+# split into parts
+parts <- strsplit(sysreqs, ", ")[[1]]
+
+# identify which is the rustc
+rustc_ver <- parts[grepl("rustc", parts)]
+
+# perform checks for the presence of rustc and cargo on the OS
+no_cargo_msg <- c(
+  "----------------------- [CARGO NOT FOUND]--------------------------",
+  "The 'cargo' command was not found on the PATH. Please install Cargo",
+  "from: https://www.rust-lang.org/tools/install",
+  "",
+  "Alternatively, you may install Cargo from your OS package manager:",
+  " - Debian/Ubuntu: apt-get install cargo",
+  " - Fedora/CentOS: dnf install cargo",
+  " - macOS: brew install rustc",
+  "-------------------------------------------------------------------"
+)
+
+no_rustc_msg <- c(
+  "----------------------- [RUST NOT FOUND]---------------------------",
+  "The 'rustc' compiler was not found on the PATH. Please install",
+  paste(rustc_ver, "or higher from:"),
+  "https://www.rust-lang.org/tools/install",
+  "",
+  "Alternatively, you may install Rust from your OS package manager:",
+  " - Debian/Ubuntu: apt-get install rustc",
+  " - Fedora/CentOS: dnf install rustc",
+  " - macOS: brew install rustc",
+  "-------------------------------------------------------------------"
+)
+
+# Add {user}/.cargo/bin to path before checking
+new_path <- paste0(
+  Sys.getenv("PATH"),
+  ":",
+  paste0(Sys.getenv("HOME"), "/.cargo/bin")
+)
+
+# set the path with the new path
+Sys.setenv("PATH" = new_path)
```

Comment:
I see that this is called both from `configure` and `configure.win`. Is that so? If yes, `":"` is an invalid `PATH` separator and `"/.cargo/bin"` is also not useful.

Reply:
This is taken from the previous `configure` and `configure.win` and is used in the `Makevars` file https://github.com/extendr/rextendr/blob/0359070b1a57116e39523209fe4345f689fbccdf/inst/templates/Makevars#L19